### PR TITLE
fix: Add TLS support, and allow changing TLS backend.

### PR DESCRIPTION
**Pull request by airblast-dev**
With the default features being disabled for the `reqwest` dependency in this commit https://github.com/LemmyNet/lemmy-client-rs/commit/398e498f458d9754d3bdc6f16714f561dfcb642e, a scheme error is raised by the request client when sending any request with `ClientOptions.secure` set to `true`. 

This enables TLS support by default and allows changing the backend via features.

### CODE REVIEW

**Comment by airblast-dev** [comment]
The workflow is failing due to `pkg-conf`, and `libssl-dev` being missing.

We can either use `reqwest/rustls-tls`, and disable `reqwest/native-tls` since `rustls-tls` doesn't require the packages mentioned above. Or we can simply install said packages. 

I will take a look into how `.woodpecker.yaml` works, and install said packages. This is likely the better solution as the default reqwest features are likely what people would expect. And where that isn't what they want they can simply change the features and use `reqwest/rustls-tls` package instead.

**Comment by SleeplessOne1917** [comment]
Thanks for opening this. I've been using this library exclusively for local development, so I hadn't been in a situation where I could encounter this issue.

**Comment by airblast-dev** [comment]
Simplest solution is going to be to just install the packages before testing. If anyone has the interest they can use a workspace and do a few tricks to optimize it.

I have spent quite a bit of time trying to set up local instances to test around with the workflow but to no success.

**Comment by SleeplessOne1917** [comment]
It could be worth just sticking to 1 way of doing TLS so it doesn't rely on having a specific package installed. If I understand correctly, rustls lets us do that?

**Comment by airblast-dev** [comment]
Thats correct and definitely an option. However I think keeping the default reqwest TLS backend is a better idea as it is likely to get better testing. The native solution also is generally a battle hardened backend which is used widely in many applications.

Either way its your choice. :+1:

**Comment by SleeplessOne1917** [comment]
Default should do the trick, though I'm having trouble finding a list of required system dependencies for it across distros/operating systems.

**Comment by airblast-dev** [comment]
I agree that is a bit problematic. I would recommend reading https://github.com/seanmonstar/reqwest/issues/2025 as they mention similar issues on the rustls side as well. When it comes to system dependencies the failed build does give out an appropriate error message for both dependencies in the case they are missing. It is also worth mentioning that most systems will not need an extra step anyway. This includes the older test image with Alpine. 

In short I think sticking with the default and providing options is the way to go considering both have various reasons to prefer one over the other.

**Review by SleeplessOne1917 — APPROVED** [review]
